tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitAssetsConfiguration.cs

```diff
@@ -25,5 +34,62 @@ public class GitAssetsConfiguration : AssetsConfiguration
         /// The auto-commit branch.
         /// </summary>
         public string AssetsRepoBranch { get; set; }
+
+        /// <summary>
+        /// The location of the assets repo for this config.
+        /// </summary>
+        public string AssetsRepoLocation { get
+            {
+                return ResolveAssetRepoLocation(true);
+            }
+        }
+
+        /// <summary>
+        /// Used to resolve the location of the "assets" store location. This is the folder CONTAINING other cloned repos. No git data will be restored or staged directly within this folder.
+        /// </summary>
+        /// <param name="autoCreate"></param>
+        /// <returns></returns>
+        public string ResolveAssetsStoreLocation(bool autoCreate = true)
+        {
+            var location = Environment.GetEnvironmentVariable("PROXY_ASSETS_FOLDER") ?? Path.Join(RepoRoot, ".assets");
+            if (!Directory.Exists(location) && autoCreate)
+            {
+                Directory.CreateDirectory(location);
+            }
+
+            return location;
+        }
+
+        /// <summary>
+        /// Resolves the location of the actual folder containing a cloned repository WITHIN the asset store. Git data will be stored directly within this directory.
+        /// </summary>
+        /// <param name="autoCreate"></param>
+        /// <returns></returns>
+        public string ResolveAssetRepoLocation(bool autoCreate = true)
+        {
+            var assetsStore = ResolveAssetsStoreLocation(autoCreate: autoCreate);
+
+            var hash = Convert.ToHexString(SHA1.Create().ComputeHash(Encoding.UTF8.GetBytes(AssetsJsonRelativeLocation)));
+
+            var location = Path.Join(assetsStore, AssetsRepo.Replace("/", "-"), hash);
+            if (!Directory.Exists(location) && autoCreate)
+            {
+                Directory.CreateDirectory(location);
+            }
+
+            return location;
+        }
+
+        /// <summary>
+        /// Checks for whether or not the assets repo is initialized.
+        /// </summary>
+        /// <param name="autoCreate"></param>
+        /// <returns></returns>
+        public bool IsAssetsRepoInitialized(bool autoCreate = true)
```

Comment:
This appears unused unless the github review is hiding things from me.

Reply:
I will remove the `autoCreate=true`, but I do need a way to know if I need to git init, sparse checkout, etc. So I'm going to leave the function, but the signature will change.